plugins/user/joomla/src/Extension/Joomla.php

```diff
@@ -107,15 +97,17 @@ public function onUserAfterDelete($user, $success, $msg): void
         $userId = (int) $user['id'];
 
         // Only execute this if the session metadata is tracked
-        if ($this->app->get('session_metadata', true)) {
+        if ($this->getApplication()->get('session_metadata', true)) {
             UserHelper::destroyUserSessions($userId, true);
         }
 
+        $db = $this->getDatabase();
+
         try {
-            $this->db->setQuery(
-                $this->db->getQuery(true)
-                    ->delete($this->db->quoteName('#__messages'))
-                    ->where($this->db->quoteName('user_id_from') . ' = :userId')
+            $this->getDatabase()->setQuery(
```

Comment:
```suggestion
            $db->setQuery(
```
Why do you not use here `$db`?